projects/plutus/test/test_current_sense.c

```diff
@@ -0,0 +1,67 @@
+#include <inttypes.h>
+
+#include "current_sense.h"
+
+#include "gpio.h"
+#include "interrupt.h"
+#include "log.h"
+#include "ltc2484.h"
+#include "soft_timer.h"
+#include "test_helpers.h"
+#include "unity.h"
+
+// Arbitrary number of testing samples
+#define TEST_NUM_SAMPLES 5
+#define TEST_INPUT_VOLTAGE 1000
+
+static volatile uint8_t s_callback_runs = 0;
+
+static CurrentSenseStorage s_storage = { 0 };
+static CurrentSenseCalibrationData s_data = { .zero_point = { 0, 0 }, .max_point = { 1000, 3 } };
+
+static void prv_callback(int32_t current, void *context) {
+  s_callback_runs++;
+
+  int32_t x_min = s_data.zero_point.voltage;
+  int32_t y_min = s_data.zero_point.current;
+  int32_t x_max = s_data.max_point.voltage;
+  int32_t y_max = s_data.max_point.current;
+
+  int32_t test_current = (y_max) * (TEST_INPUT_VOLTAGE - x_min) / (x_max - x_min);
```

Comment:
You're not really testing much. Do something simple like if your bounds are 0,0 and 1000,10 then 5 should be 500 and -5 should be -500.

Reply:
Fixed